### ISSUE TITLE
CodeReader 0.7.11 seems to "hang on first scan" under SailfishOS 2.1.1

### ISSUE THREAD

**Issue by Olf0** [steffen-foerster/sailfish-barcode#33]
CodeReader 0.7.11 seems to expose a "hanging on first scan" bug under SailfishOS 2.1.1, see https://together.jolla.com/question/166973/codereader-hanging-forever-on-jolla1/ for details. 
I can reproduce this (although I never had the patience to wait for the first scan to finish). 

Note, that CodeReader 0.7.11 worked fine (on my Jolla 1 phone) under SailfishOS 1.1.x and 2.0.x

**Comment by steffen-foerster** [comment]
I cannot reproduce this behaviour. Here is my testing environment:

- Jolla 1
- CodeReader version 0.7.11
- SailfishOS 2.1.1.26
- Scanned QR code:  https://de.wikipedia.org/wiki/QR-Code

Each scan (including the first) needs only 1-2 seconds.

**Comment by Olf0** [comment]
Oops: 
@steffen-foerster, your comment caused me to retest with some QR-Codes on magazines (I have used only classic 1D-Barcodes for testing this issue, so far): 
Working (i.e. no hang on first run, but scanning garbage, which is not scope of this issue, here).

So I went back to trying 1D-Barcodes: 

- A shaving foam bottle, with high contrast 1D-Barcode print in good lighting: Working absolutely fine.

- Back to the Coca-Cola bottle (out of thin, floppy PET), I originally tried: Fails to recognise the Barcode (probably due to heavy reflections), but no hang.  Scans are abortable now and everything works as it did with CodeReader 0.7.11 under older releases of SailfishOS. 

On first sight, it looks, as if CodeReader needs a single successful Barcode scan to function properly thereafter. 

Sorry to have bothered you with this (apparently) almost non-issue. 
Hence closing this issue and reporting back to TJC.

**Comment by Olf0** [comment]
@steffen-foerster, I missed to emphasise how much I appreciate CodeReader as the only well working, fast 1D- and 2D-Barcode scanning app for SailfishOS.  Although the Android apps QR Scanner and Barcode Scanner support more types of Barcodes and provide a higher detection rate, the are not accessible for all SailfishOS users (not for those without AlienDalvik) and slower at start-up, due to being Android apps. 

Kudos to you for this nice software!

**Comment by steffen-foerster** [comment]
@Olf0, thanks for retesting and closing this issue. I'm glad my app is still useful :-)

**Comment by prometheosgh** [comment]
@Olf @steffen-foerster I started the question https://together.jolla.com/question/166973/codereader-hanging-forever-on-jolla1/ which gave origin tho this thread.
- Unfortunately i did not write up which QR-codes I used for testing before opening this Question on together/jolla. So I repeat here my testing.
- I use now the QR-Codes from the site https://en.wikipedia.org/wiki/QR_code. I use the codes as they appear as a slide show, after pointing to the first QR, and in that precise order.
- They are 20 2D QR-codes and one 1D Bar Code.
- 4 codes failed now in my test (Japanese immigration landing permission..., Micro QR, 4-color HCC2D code and 8-color HCC2D code. )
- All other 17 codes passed the test.
- Nevertheless I remember clearly on my first testing my QR-codes and the message "Scan in progress..." lasting for minutes.
- But now also some "successful" codes could let me think they failed: like "Version 40 (177×177)" which takes now 40 s before telling me "Failed, retry." and on repeating succeeds after 20 s. Most of the codes bring success after less than one second.
In conclusion, for my first test I cannot exclude having mistaken some false negative test. Therefore I **WITHDRAW the original QUESTION**. I think it is important to keep only clear facts in statements in the Internet. 
For some days I will keep the original question for reference for You and for interested users, and eventually delete it.
Regards prometheos(gh)

PS: the longer I use codereader the more I like it. First of all: it will never beams you up to some bad site before you see what it has read. Not bad.
